### PR TITLE
bmi160: Fix I2C support

### PR DESCRIPTION
**Pull request by avigreen1978**
bmi160 driver code supports i2c, but is not selectable by Kconfig
Also, previous yaml supported only spi configuration, so added support for both, depending on dts

Signed-off-by: Avi Green <avigreen1978@yandex.com>

### CODE REVIEW

File: dts/bindings/sensor/bosch,bmi160-i2c.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2018, Linaro Limited
+# SPDX-License-Identifier: Apache-2.0
+
+description: BMI160 inertial measurement unit
+
+compatible: "bosch,bmi160"
+
+include: [i2c-device.yaml, "bosch,bmi160.yaml"]
```
Comment (bbilas) [inline dts/bindings/sensor/bosch,bmi160-i2c.yaml:8]: Missing nl.

File: dts/bindings/sensor/bosch,bmi160-i2c.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2018, Linaro Limited
```
Comment (bbilas) [inline dts/bindings/sensor/bosch,bmi160-i2c.yaml:1]: Actually it's 2022 ;)

**Review by mbolivar-nordic — APPROVED** [review]
This looks right to me. @sjg20 how have you been using this driver with I2C without these bindings?

**Review by MaureenHelm — DISMISSED** [review]

**Comment by bbilas** [comment]
@avigreen1978 You should rebase with the main branch.

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
Yes, this needs to be rebased instead of having a merge commit

**Comment by avigreen1978** [comment]
I believe the failed check has nothing to do with my PR

**Review by teburd — DISMISSED** [review]

**Comment by MaureenHelm** [comment]
> I believe the failed check has nothing to do with my PR

Please rebase to pick up the fix in #43447

**Review by teburd — APPROVED** [review]

**Comment by MaureenHelm** [comment]
@mbolivar-nordic please revisit

**Review by mbolivar-nordic — APPROVED** [review]

**Pull request merged**